### PR TITLE
fix: read float param

### PR DESCRIPTION
**Pull request by storrrrrrrrm**
Signed-off-by: suchang <chang.su@autocore.ai>

## Description

before:
屏幕截图

after:
image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by taikitanaka3** [comment]
@storrrrrrrrm 
is this a fix PR or you want to restrict value to float in this PR?
I think before cases include  node_->declare_parameter<float>("z_offset", 2); this change so it should die.
but I think if this change is not included z_offset is read int and cast to float, right?

**Comment by storrrrrrrrm** [comment]
> 

it is a fix,z_offset is defined as a float value, but if you config z_offset=-2.0 in launch.xml,when you 
```
ros2 launch lidar_apollo_instance_segmentation lidar_apollo_instance_segmentation.launch.xml
```

it crash

**Comment by taikitanaka3** [comment]
@storrrrrrrrm 
image
The sign of z offset is reversed , Do you know better default value? It should be positive or negative both.

File: perception/lidar_apollo_instance_segmentation/src/detector.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ LidarApolloInstanceSegmentation::LidarApolloInstanceSegmentation(rclcpp::Node *
   use_intensity_feature = node_->declare_parameter("use_intensity_feature", true);
   use_constant_feature = node_->declare_parameter("use_constant_feature", true);
   target_frame_ = node_->declare_parameter("target_frame", "base_link");
-  z_offset_ = node_->declare_parameter("z_offset", 2);
+  z_offset_ = node_->declare_parameter<float>("z_offset", 2);
```
Comment (taikitanaka3) [inline perception/lidar_apollo_instance_segmentation/src/detector.cpp:41]:
```suggestion
  z_offset_ = node_->declare_parameter<float>("z_offset", -2.0);
```

**Comment by storrrrrrrrm** [comment]
> @storrrrrrrrm image The sign of z offset is reversed , Do you know better default value? It should be positive or negative both.

-2 is better for default value when you set target_frame='base_link' in launch.xml,  it means the height from target_frame to a fake toplidar frame which i think the apollo model trained from.

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**